### PR TITLE
Added support for filtering incidents by statuses

### PR DESCRIPTION
**Pull request by mgpaja8**
Things to talk about:
1) Discuss parsing the query (I implemented simple one)
2) Add support to filter incidents by severity and related task status

Please review what i did and we can maybe setup a quick call for tomorrow, just to make sure i am on right path.

### CODE REVIEW

File: tests/QualityKitchen.Application.Tests.Functional/EventHandlers/CreateIncidentHandlerTests.cs [hunk 1]
```diff
@@ -171,6 +171,11 @@ public IEnumerable<Incident> GetAll(Guid operationId)
                 return this._incidents;
             }
 
+            public IEnumerable<Incident> GetFilteredByStatus(Guid operationId, string status)
```
Comment (j0nscalet) [inline tests/QualityKitchen.Application.Tests.Functional/EventHandlers/CreateIncidentHandlerTests.cs]: This isn't the right spot to functional tests this stuff. You'll want to write your tests in EmployeeScenarios. They key here is to think about the use case that you're testing. In this case it's an Employee (or the system) that "fetches all open incident", where 'open' is a certain status.

File: src/QualityKitchen.Application/ApplicationServices/IncidentApplicationService.cs [hunk 2]
```diff
@@ -15,7 +15,6 @@ public class IncidentApplicationService : IIncidentApplicationService
     {
         private IIncidentRepository _incidentRepository;
         private IOperationRepository _operationRepository;
-        private IChecklistRepository _checklistRepository;
```
Comment (mgpaja8) [inline src/QualityKitchen.Application/ApplicationServices/IncidentApplicationService.cs]: this was not being used

File: src/QualityKitchen.RestAPI/Controllers/IncidentController.cs [hunk 3]
```diff
@@ -84,5 +85,49 @@ public IActionResult Resolve()
...
+            ServiceResult serviceResult = this._incidentApplicationService.IncidentsBy(operationGuid, incidentStatus);
+
+            if (serviceResult.ResultCode != ServiceResult.ServiceResultCode.Success)
+                return this.ErrorResult(serviceResult);
+
+            return this.SuccessResult(serviceResult);
+        }
+
+        [HttpGet("operations/{operationId}/incidents/?severity={severity}")]
+        public IActionResult FilterBySeverity([FromQuery]string operationId, [FromQuery]string severity)
```
Comment (j0nscalet) [inline src/QualityKitchen.RestAPI/Controllers/IncidentController.cs]: Overall this is seriously great work. Code is super clean! Feels good doesn't it? 😎 

Now you might begin to understand where the 'dynamic query'  from the query string idea came from. So here's a question what if I wanted to filter by both status and severity like this:
```
/operations/{id}/incidents/?status=active&severity=minor
```

It's a hypothetical and isn't necessary for this issue, what do you think? Smells like something that's been solved before by other people too. Can you do some research and see what you find?

File: src/QualityKitchen.Domain/Interfaces/IIncidentRepository.cs [hunk 4]
```diff
@@ -2,16 +2,18 @@
 using System.Collections.Generic;
 
 using QualityKitchen.Domain.Model;
+using static QualityKitchen.Domain.Model.Incident;
```
Comment (j0nscalet) [inline src/QualityKitchen.Domain/Interfaces/IIncidentRepository.cs]: Why is this here?

File: src/QualityKitchen.Repositories.EfCore/Repositories/IncidentRepository.cs [hunk 5]
```diff
@@ -72,6 +73,28 @@ public DomainModel.Incident Find(Func<DomainModel.Incident, bool> predicate)
         {
             throw new NotImplementedException();
         }
+
+        public IEnumerable<DomainModel.Incident> FindBy(Guid operationId, IncidentStatus status)
```
Comment (j0nscalet) [inline src/QualityKitchen.Repositories.EfCore/Repositories/IncidentRepository.cs]: Instead of the static reference you should be able to do:
```DomainModel.Incident.IncidentStatus status)```
here

**Comment by j0nscalet** [comment]
@mgpaja8 good teamwork on this PR man. High five ✋!

Take a minute and review the changes I made (if you haven't already) so you're familiar.
